Enhance modifier + num launching

gdziedzic opened the pull request:
# Enhance modifier + num launching

* Adding the ability to set custom modifiers in settings
![image](https://user-images.githubusercontent.com/36335367/80921183-1a4a7a00-8d75-11ea-819a-f8082a35d5e8.png)
* as a result user see this
![image](https://user-images.githubusercontent.com/36335367/80921195-32ba9480-8d75-11ea-833b-2a8468fe3327.png)

# Next steps
* in the future we might show the exact hotkey on the list (instead of _Mod+<num>_ . I haven't added it right now because it's a bit more complex and I wanted to get feedback from the users whether it doesn't make it too cluttered. 
* alternatively we could step back and take a more minimalistic approach and display just the number
![image](https://user-images.githubusercontent.com/36335367/80921341-19661800-8d76-11ea-8a1a-652eeb5d9332.png)

JohnTheGr8 commented:
Haven't looked at the code yet, but this looks nice.

> in the future we might show the exact hotkey

Agreed.

> alternatively we could step back and take a more minimalistic approach and display just the number

Suggestion: we should let users completely hide this if they are comfortable with the shortcuts.

SysC0mp commented:
Great addition, love it. I was also able to test it without any issues. 👍 

> in the future we might show the exact hotkey

That would be great because the Mod+[1-9] could confuse users. We could also make this a part of each theme so that users could make it larger / smaller like they want to. At the moment I think it is a bit too big, but that is completely personal preference.

> Suggestion: we should let users completely hide this if they are comfortable with the shortcuts.

I also agree with that because I think it is always the best option to let the user decide how they want it to behave.

JohnTheGr8 commented:
> At the moment I think it is a bit too big

Agreed, perhaps it should be same font size as the subtitle and aligned with it by default?

jjw24 commented on Flow.Launcher/ViewModel/SettingWindowViewModel.cs:
```diff
@@ -139,6 +139,7 @@ public List<string> QuerySearchPrecisionStrings
             }
         }
 
+        public List<string> OpenResultModifiersList => new List<string> { "Alt", "Ctrl", "Ctrl+Alt" };
```
great addition @gdziedzic.  Nitpick here, could we put the strings in Constant? I see them referenced more than once.

Also as @JohnTheGr8 mentioned, could we add an option to hide the modifier key display on main window please

gdziedzic commented:
> Also as mentioned, could we add an option to hide the modifier key display on main window please
![image](https://user-images.githubusercontent.com/36335367/80926349-0adc2880-8d97-11ea-81ba-bc9fb86a9333.png)

* Added ability to hide
* Displaying full hotkey
![image](https://user-images.githubusercontent.com/36335367/80926378-4119a800-8d97-11ea-8dcb-b63e6e04fa8a.png)
* changed font style to subitem as suggested
* extracted common consts

SysC0mp commented:
Looking really nice now, great work! 👍 

One thing I stumbled upon while testing: 
If I set the "Maximum results shown" to something higher than 9, the modifier will still be shown but cannot be used. I guess the maximum shown modifier should be 9. 

![image](https://user-images.githubusercontent.com/8371978/80926929-8c818580-8d9a-11ea-9a0a-aac3079442ba.png)

gdziedzic commented:
> , the modifier will still be shown but cannot be used. I guess the maximum shown modifier should be 9.

Good spot thank you! Fixed now

![image](https://user-images.githubusercontent.com/36335367/81002942-5dc2e800-8e4a-11ea-855b-0918829a7fbf.png)

SysC0mp commented:
Thank you, looking fine to me now 🥳
